flight/PiOS/Common/pios_sensors.c

```diff
@@ -31,60 +31,93 @@
 #include <stddef.h>
 
 //! The list of queue handles
-static struct pios_queue *queues[PIOS_SENSOR_LAST];
-static uint32_t sample_rates[PIOS_SENSOR_LAST];
-static int32_t max_gyro_rate;
+typedef bool (*PIOS_SENSOR_Callback_t)(void *ctx, void *output,
+		int ms_to_wait, int *next_call);
+
+static struct PIOS_Sensor {
+	PIOS_SENSOR_Callback_t cb_func;
```

Comment:
Can we have a more descriptive name for this please? e.g. `get_data_cb`?

Reply:
As you wish.